eth/backend.go

```diff
@@ -363,6 +350,37 @@ func makeExtraData(extra []byte) []byte {
 	return extra
 }
 
+func initializeL2BaseFeeCoefficients(chainConfig *params.ChainConfig, blockchain *core.BlockChain) error {
+	state, err := blockchain.State()
+	if err != nil {
+		return err
+	}
+
+	l2BaseFeeOverhead := common.Big0
+	l2BaseFeeScalar := common.Big0
+
+	if l2SystemConfig := chainConfig.Scroll.L2SystemConfigAddress(); l2SystemConfig != (common.Address{}) {
+		l2BaseFeeOverhead = state.GetState(l2SystemConfig, rcfg.L2BaseFeeOverheadSlot).Big()
+		l2BaseFeeScalar = state.GetState(l2SystemConfig, rcfg.L2BaseFeeScalarSlot).Big()
+	} else {
+		log.Warn("L2SystemConfig address is not configured")
+	}
+
+	// fallback to default if contract is not deployed or configured yet
+	if l2BaseFeeOverhead.Cmp(common.Big0) == 0 {
```

Comment:
Maybe also add a log here indicating that the contract is added in the config (deployed or address is predicted), but not having meaningful params (e.g., not initialized both params) yet.

Reply:
Another point is maybe it's more convenient to combine `updateBaseFeeOverhead` and `updateBaseFeeScalar` in the contract. e.g., (i) retrieve one event log, (ii) allow one of 0s (l2 base fee overhead and l2 base fee scalar) in corner case settings, (iii) and they are often updated in the meantime.

One of the cons might be extra gas overhead to update only one param, but this case is rare historically.

Reply:
Yes. My point is currently due to separating into 2 update functions (`updateBaseFeeOverhead` and `updateBaseFeeScalar`), one of them `0` would be an "uninitialized" state, but if merging into 1 function, only two of them `0` would be an "uninitialized" state. And also normally we update these two params in the meantime, no?

Reply:
Yes, but if any of them is uninitialized here, we will use the fallback value for it.

I don't see a huge difference in two contract methods/evens or just one. I'd rather keep as is since https://github.com/scroll-tech/scroll-contracts/pull/114 is merged, unless you can foresee any issue with this.

Reply:
Not found any issues, while a small point would be if updating the two params, should combine them into a batch call. Or there's a small time window that only one param is updated.